lib/cdo/delete_accounts_helper.rb

```diff
@@ -126,18 +126,23 @@ def self.anonymize_user_sections(user_id)
   # (reporting only).
   # @param [Integer] The user ID to purge from Pardot.
   def self.remove_from_pardot(user_id)
-    pardot_ids = PEGASUS_DB[:contact_rollups].
-      select(:pardot_id).
-      where(dashboard_user_id: user_id).
-      map {|contact_rollup| contact_rollup[:pardot_id]}
-    failed_ids = Pardot.delete_prospects(pardot_ids)
-    if failed_ids.any?
-      raise "Pardot.delete_prospects failed for Pardot IDs #{failed_ids.join(', ')}."
+    # Though we have the DB tables in all environments, we only sync data from the production
+    # environment with Pardot.
+    if rack_env == :production
+      pardot_ids = PEGASUS_DB[:contact_rollups].
+        select(:pardot_id).
+        where(dashboard_user_id: user_id).
+        map {|contact_rollup| contact_rollup[:pardot_id]}
+      failed_ids = Pardot.delete_prospects(pardot_ids)
+      if failed_ids.any?
+        raise "Pardot.delete_prospects failed for Pardot IDs #{failed_ids.join(', ')}."
```

Comment:
This is going to fail once in a while. Pardot will occasionally, although not very often, time out or return a 5xx or 4xx response. What are the ramifications if it fails? If it leaves this user hanging around partially unpurged but we will try again for that user on the next cron run, that's fine. Or could it potentially leave some data orphaned and have us not try again?